agent/php_wrapper.c

```diff
@@ -7,20 +7,47 @@
 #include "php_user_instrument.h"
 #include "php_wrapper.h"
 #include "util_logging.h"
+static char* Z_TYPE_STR(zval* zv) {
+
+  switch (Z_TYPE_P(zv)) {
+    case IS_NULL:
+      return "IS_NULL";
+
+    case IS_STRING:
+      return "IS_STRING";
+
+    case IS_LONG:
+      return "IS_LONG";
+
+#if ZEND_MODULE_API_NO >= ZEND_7_0_X_API_NO /* PHP7+ */
+    case IS_TRUE:
+      return "IS_TRUE";
+
+    case IS_FALSE:
+      return "IS_FALSE";
+#else
+    case IS_BOOL:
+      return "IS_BOOL";
+#endif /* PHP7 */
+
+    case IS_DOUBLE:
+      return "IS_DOUBLE";
+
+    case IS_OBJECT:
+      return "IS_OBJECT";
+
+    case IS_ARRAY:
+      return "IS_ARRAY";
+
+    default:
+      return "UNKNOWN";
+  }
+}
```

Comment:
I'm not sure about this...

Reply:
I like it alright, though I think it would be beneficial to rename the function. With its current name, I would assume it is provided by the PHP internals API. It might also be worth mentioning that this is (I think) currently only used for logging. Maybe put "log" somewhere in the name so that somebody has to justify a rename for a more important use-case.

Reply:
I wonder if it might be better to make `nr_format_zval_for_debug` function external to `agent/php_execute.c`. I sometimes wish it was as it makes debugging so much easier...

Reply:
I agree with @ZNeumann's comment regarding the naming convention of this function.

Reply:
@ZNeumann @bduranleau-nr What do you think about the idea of making `nr_format_zval_for_debug` function external to `agent/php_execute.c` and using it instead of adding `Z_TYPE_STR`?

Reply:
works for me

Reply:
agree 👍

Reply:
@bduranleau-nr, @ZNeumann: See e3938763a55965361a99778e52e94578dc382b8f.